lab-kyle/lib/client-pool.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+
+ // register socket(clients) to the pool
+function registerClient(socket, clientPool){
+  socket.wack = {};
+  socket.wack.id = 'client_' + Date.now();
```

Comment:
You may want to consider using the UUID library (node-uuid) instead of Data.now() since, theoretically, Date.now() could assign the same value to multiple users.